parser/prism/Translator.cc

```diff
@@ -499,6 +505,27 @@ std::unique_ptr<parser::Node> Translator::translate(const Node &node) {
     return translate(node.get_raw_node_pointer());
 }
 
+// The legacy Sorbet parser doesn't have a counterpart to PM_ARGUMENTS_NODE to wrap the array
+// of argument nodes. It just uses a NodeVec directly, which is what this function produces.
+NodeVec Translator::translateArguments(pm_arguments_node *argsNode, size_t extraCapacity) {
+    NodeVec results;
+
+    if (argsNode == nullptr) {
+        results.reserve(extraCapacity);
```

Comment:
Nit: I don't have strong opinions but it looks like we're adding a few lines for `extraCapacity` reserve logic but only inserting once on line 84. I'm not sure it's worth ternary operator, extra argument and this reserve call. Is there a performance improvement in practice to justify it?

Reply:
Yeah that insertion on line 84 can trigger a copy of the whole vec, which I'm trying to avoid.

I haven't profiled this in practice (we can't parse a large enough program for it to matter yet :D), but "avoid allocations where it's easy to do so" usually falls in the "meh, worth it" bucket

I'd be happy to come back and measure this once we can parse "real" programs